apm-agent-plugins/apm-urlconnection-plugin/src/main/java/co/elastic/apm/agent/urlconnection/HttpUrlConnectionInstrumentation.java

```diff
@@ -79,15 +79,19 @@ public static class AdviceClass {
             @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
             public static Object enter(@Advice.This HttpURLConnection thiz,
                                        @Advice.FieldValue("connected") boolean connected,
+                                       @Advice.FieldValue("responseCode") int responseCode,
                                        @Advice.Origin String signature) {
 
+                //With HEAD requests the connected stays false
```

Comment:
[minor] that's really an implementation detail here, could we also check the HTTP verb to be sure ?

Reply:
I'd argue that even for non-HEAD requests this approach is safer: If a responseCode is available, this implies that the connection has already been made. Therefore I think an additional check would not be good here.

Reply:
Good for me, let's keep it as-is then !